### DISCUSSION TITLE
Assigning alias [You don't have access to the domain]

### DISCUSSION THREAD

**Discussion by shpsyte** [vercel/community#1826] [Help] [answered]
### Summary

Our account is under a team and we're trying to add a specific URL to our preview environment

But when I run the code 

vercel alias set [deployment-url] [custom-domain] --token=<my_token> --scope=<team_token>

I got an error 

Error: You don't have access to the domain [custom-domain] under [team-name]

- the domain already have its A and CNAME pointed on correctly 
- I'm the owner of the team 

Vercel Team:
This is the DeploymentId : UGu2BEnfiWtFx5FVi22GPHecjwuH

Community Team
Any Idas what had happened, or how can I fix this?

**Comment by shpsyte** [comment, marked as answer]
Found it.

Turns out you need to add the domain here, and then run the cli


https://vercel.com/dashboard/<your-team-id>/domains

**Reply by adams-family** [reply]
Doesn't this assign the domain to the Production version?

**Reply by sagspot** [reply]
And how do you prevent this domain from being assigned to production deployments?

**Reply by johnymanuelli** [reply]
Any update on how to add a custom domain but not linking it automatically to a production deployment ? I dont want to link all my git projects just because i need 1 custom preview url.

**Reply by nathanielpomianowski** [reply]
**UPDATE**
For teams you need to add the `--scope`.
i.e.: `vercel alias set deployment-url <big-plans-dev-domain> --token=$VERCEL_TOKEN --scope=BigPlans`


 It would seem the a token with the team scope should be efficient enough for this when the team has the correct domain set, however, I am still getting this error.

Perhaps something was missed when developing this? Here is the route where the teams domain is set: `https://vercel.com/dashboard/scope-name/domains`

i.e.
Personal Account
- FredtheDev

Teams
- BigPlans

When you try to use the command:
`vercel alias set deployment-url <big-plans-dev-domain> --token=$VERCEL_TOKEN`

The follow error occurs:

> Fetching domain `<big-plans-dev-domain>` under FredtheDev
> Error: You don't have access to the domain `<big-plans-dev-domain>` under FredtheDev.

**Reply by kmkinney** [reply]
Another thing I discovered is that if you don't add the https:// before your custom domain in this command, you will get the `Error: You don't have access` issue.

However, if you do add the https:// before the custom domain it looks like it would work, but in my case it errors after a little bit with this message
```
Error: An unexpected error occurred in alias: Error: Response Error (508)
```

**Reply by dnzg** [reply]
Any updates on this topic? Still have the same problems.

**Comment by nathan-ahn** [comment]
For me, the issue was that a personal project still had the domain assigned to it (example.com) while I was trying to alias with a subdomain (abc.example.com). If you're trying to add to https://vercel.com/dashboard/domains, there's a chance there's a conflicting higher level domain in another team of yours.

**Reply by charanjit-singh** [reply]
Can you give example?

**Reply by illuisionary** [reply]
How did you solve it? For me, it doesn't work when it's there or when it's absent.

**Reply by dnzg** [reply]
Same for me, so I had to go with default vercel.app domain temporarily.

**Reply by danielzting** [reply]
We (me and @qyuo) solved it by specifying a scope with the `--scope` flag. You can see a list of available scopes by doing `vercel switch`. In our case we only had one and using that worked but if you have multiple you might need to do some trial and error. I don't know what Vercel changed about their whole teams structure/infra/CLI (I noticed on the web UI my projects were forcibly transferred into a new free team when they were before in my personal account, and there seems to be no way to move them out of a team back into just an account) because I never had to do this before until one day our CI/CD pipelines just started failing due to this error.

**Reply by dnzg** [reply]
I tried using `--scope`, but Vercel gives this error, saying that none of my scopes have access to the custom domain, that is assigned to my project.

**Comment by uhrohraggy** [comment]
What helped me was calling `vercel switch`, choosing the team, and you should receive `> Success! The team Foo (foo) is now active!`

Then calling `vercel alias thisurl.foo.com atdeployment-stub-teamfoo.vercel.app` without https protocol.

**Comment by chirag-ema** [comment]
Need a solution on this. I have domains under my "project" but not in my "team". I want to set alias after doing a deploy via github action. But running into the above mentioned issues (you don't have access to the <domain> under <project>).
